File: Vigilantes.DaprWorkshop.ReceiptGenerationService/node/controllers/ReceiptGenerationController.js
const Receipt = require('../models/Receipt');

// POST /orders
exports.generateReceipt = async function(req, res) {
    const orderSummary = req.body;
    console.log(`Created Order Summary: OrderId = ${orderSummary.orderId}, StoreId = ${orderSummary.storeId}, CustomerName = ${orderSummary.customerName}, LoyaltyId = ${orderSummary.loyaltyId}, OrderItemCount = ${orderSummary.orderItems.length}, OrderTotal = ${orderSummary.orderTotal}`);

    // TODO: Challenge 5 - Store receipt via a Dapr Output Binding that can be used as a data sink
    res.json();
}

File: Vigilantes.DaprWorkshop.ReceiptGenerationService/csharp/Controllers/ReceiptGenerationController.cs
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;
using CloudNative.CloudEvents;
using Dapr;
using Dapr.Client;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json.Linq;
using Vigilantes.DaprWorkshop.ReceiptGenerationService.Models;

namespace Vigilantes.DaprWorkshop.ReceiptGenerationService.Controllers
{
    [ApiController]
    public class ReceiptGenerationConsumerController : ControllerBase
    {
        private readonly HttpClient _httpClient;
        private readonly ILogger<ReceiptGenerationConsumerController> _logger;

        private readonly DaprClient _daprClient;

        public ReceiptGenerationConsumerController(IHttpClientFactory httpClientFactory, ILogger<ReceiptGenerationConsumerController> logger, DaprClient daprClient)
        {
            _httpClient = httpClientFactory.CreateClient();
            _logger = logger;
            _daprClient = daprClient;
        }

        [Topic("pubsub", "newOrder")]
        [Route("/orders")]
        public async Task<IActionResult> GenerateReceipt(CloudEvent cloudEvent)
        {
            var orderSummary = ((JToken)cloudEvent.Data).ToObject<OrderSummary>();

            _logger.LogInformation("Writing Order Summary (receipt) to storage: {@OrderSummary}", orderSummary);
            await  _daprClient.InvokeBindingAsync("bloborders", "create", orderSummary);
            
            return Ok();
        }
    }
}
